src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -98,6 +93,68 @@ public static FunctionPointerMethodSymbol CreateFromSource(FunctionPointerTypeSy
                 typeBinder,
                 diagnostics,
                 suppressUseSiteDiagnostics);
+
+            // PROTOTYPE(func-ptr): Do this right. Just making compatible changes for now
+            static CallingConvention getCallingConvention(FunctionPointerCallingConventionSyntax? callingConventionSyntax, DiagnosticBag diagnostics)
+            {
+                switch (callingConventionSyntax?.ManagedOrUnmanagedKeyword.Kind())
+                {
+                    case null:
+                        return CallingConvention.Default;
+
+                    case SyntaxKind.ManagedKeyword:
+                        // Possible if we get a node not constructed by the parser
+                        if (callingConventionSyntax.UnmanagedCallingConventionList is object && !callingConventionSyntax.ContainsDiagnostics)
+                        {
+                            diagnostics.Add(ErrorCode.ERR_CannotSpecifyManagedWithUnmanagedSpecifiers, callingConventionSyntax.UnmanagedCallingConventionList.GetLocation());
+                        }
+                        return CallingConvention.Default;
+
+                    case SyntaxKind.UnmanagedKeyword:
+                        switch (callingConventionSyntax.UnmanagedCallingConventionList)
+                        {
+                            case null:
+                                return CallingConvention.Unmanaged;
+
+                            case { CallingConventions: { Count: 1 } specifiers }:
+                                return specifiers[0].Name switch
+                                {
+                                    { Text: "Cdecl" } => CallingConvention.CDecl,
```

Comment:
>Text [](start = 38, length = 4)

Should it be Text or ValueText? Throughout this switch. We probably should have tests with escaped identifiers.